packages/web-features/package.json

```diff
@@ -13,14 +13,17 @@
     ".": "./index.js",
     "./index.json": "./index.json"
   },
+  "types": "./index.d.ts",
   "files": [
+    "index.d.ts",
     "index.js",
     "index.json"
   ],
   "scripts": {
-    "prepare": "tsc"
+    "prepare": "tsc && rm types.js && tsup ./index.ts --dts-only --format=esm --out-dir=."
```

Comment:
Ironically, TypeScript does not give us very good tools to generate type definitions. This might change in the future (there's a `tsc` option on the roadmap for TS 5.4), but for now I found a halfway decent tool — `tsup` — that does what I need: builds one `index.d.ts` file containing everything we need.

Reply:
I've been using [`api-extractor`](https://api-extractor.com) for this.
I also wanted to have a single `.d.ts` file and found that `api-extractor` works well and also has other interesting features.

`tsup` is obviously fine aswel :)

Reply:
Thanks for highlighting this, @romainmenke. In the interest of shipping, I'll merge this as is, but `api-extractor` (and its docs) seem more approachable than `tsup`. If I ever have to make changes to how this works, I'll revisit swapping tools then.